pennylane/devices/default_qutrit.py

```diff
@@ -0,0 +1,258 @@
+from pennylane import QutritDevice
+from pennylane.wires import WireError  # pylint: disable=unused-import
+from .._version import __version__
+
+# tolerance for numerical errors
+tolerance = 1e-10
+
+OMEGA = np.exp(2 * np.pi * 1j / 3)
+
+
+class DefaultQutrit(QutritDevice):
+    """Default qutrit device for PennyLane.
+
+    Args:
+        wires (int, Iterable[Number, str]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers (i.e., ``[-1, 0, 2]``)
+            or strings (``['ancilla', 'q1', 'q2']``). Default 1 if not specified.
+        shots (None, int): How many times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Defaults to ``None`` if not specified, which means that the device
+            returns analytical results.
+    """
+
+    name = "Default qutrit PennyLane plugin"
+    short_name = "default.qutrit"
+    pennylane_requires = __version__
+    version = __version__
+    author = "Mudit Pandey"
+
+    # TODO: Update list of operations and observables once more are added
+    operations = {
+        "QutritUnitary",
+    }
+
+    observables = {
+        "Identity",
+    }
+
+    def __init__(
+        self,
+        wires,
+        *,
+        r_dtype=np.float64,
+        c_dtype=np.complex128,
+        shots=None,
+        analytic=None,
+    ):
+        super().__init__(wires, shots, r_dtype=r_dtype, c_dtype=c_dtype, analytic=analytic)
+        self._debugger = None
+
+        # Create the initial state. Internally, we store the
+        # state as an array of dimension [3]*wires.
+        self._state = self._create_basis_state(0)
+        self._pre_rotated_state = self._state
+
+        # TODO: Add operations
+        self._apply_ops = {
+            # All operations that can be applied on the `default.qutrit` device by directly
+            # manipulating the internal state array will be included in this set
+        }
+
+    @functools.lru_cache()
+    def map_wires(self, wires):
+        # temporarily overwrite this method to bypass
+        # wire map that produces Wires objects
+        try:
+            mapped_wires = [self.wire_map[w] for w in wires]
+        except KeyError as e:
+            raise WireError(
+                f"Did not find some of the wires {wires.labels} on device with wires {self.wires.labels}."
+            ) from e
+
+        return mapped_wires
+
+    def define_wire_map(self, wires):
+        # temporarily overwrite this method to bypass
+        # wire map that produces Wires objects
+        consecutive_wires = range(self.num_wires)
+        wire_map = zip(wires, consecutive_wires)
+        return dict(wire_map)
+
+    def apply(self, operations, rotations=None, **kwargs):  # pylint: disable=arguments-differ
+        rotations = rotations or []
+
+        # apply the circuit operations
+        for i, operation in enumerate(operations):  # pylint: disable=unused-variable
```

Comment:
Perhaps leave a TODO here that mentions that eventually this will be extended to take into account basis state / state vector / snapshot operations like in the qubit device.